ffi/diplomat/src/collator.rs

```diff
@@ -0,0 +1,239 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[diplomat::bridge]
+pub mod ffi {
+    use alloc::boxed::Box;
+    use diplomat_runtime::DiplomatResult;
+    use icu_collator::{Collator, CollatorOptions};
+
+    use crate::{
+        errors::ffi::ICU4XError, locale::ffi::ICU4XLocale, provider::ffi::ICU4XDataProvider,
+    };
+
+    #[diplomat::opaque]
+    #[diplomat::rust_link(icu_collator::Collator, Struct)]
+    pub struct ICU4XCollator(pub Collator);
+
+    #[diplomat::enum_convert(core::cmp::Ordering)]
+    #[diplomat::rust_link(core::cmp::Ordering, Enum)]
+    pub enum ICU4XOrdering {
+        Less = 0,
+        Equal = 1,
+        Greater = 2,
+    }
+
+    #[non_exhaustive]
```

Comment:
Question: Is there a purpose of `#[non_exhaustive]` in FFI?

Reply:
not really, no

(though i can imagine Diplomat itself gaining support for handling enums slightly differently in these cases)

Reply:
Should I remove the attribute, then?  It existed on the corresponding Rust enums in the collator component, which is why I added them here.

Reply:
Yes please

Reply:
Done.